scripts/gen_ldelf_hex.py

```diff
@@ -37,46 +37,65 @@ def round_up(n, m):
 
 def emit_load_segments(elffile, outf):
     load_size = 0
+    code_size = 0
     data_size = 0
-    next_rwseg_va = 0
+    load_segments = [s for s in elffile.iter_segments()
+                     if s['p_type'] == 'PT_LOAD']
+    load_segments.sort(key=lambda s: s['p_vaddr'])
```

Comment:
Aren't the segments guaranteed to be sorted by virtual address already?

Reply:
Correct. `iter_segments()` returns the segments in the order they appear in the file (program headers) and the ELF spec specifies "Loadable segment entries in the program header table appear in ascending order, sorted on the `p_vaddr` member." (https://docs.oracle.com/cd/E19683-01/816-1386/chapter6-83432/index.html).

I'll remove the sort (altough it was a good opportunity for me to demonstrate my knowledge of Python lambda functions :laughing:).

Reply:
> I'll remove the sort (altough it was a good opportunity for me to demonstrate my knowledge of Python lambda functions ).

Haha :-)